### PR TITLE
Handle interrupts: synchronous `KeyManagementService` uses Socket IO (open, read, write), which is interruptible in a virtual thread

### PR DESCRIPTION
**Pull request by stIncMale**
This PR is very similar to https://github.com/mongodb/mongo-java-driver/pull/1203, which is why I assigned the same reviewers.

JAVA-5139

### CODE REVIEW

File: driver-sync/src/main/com/mongodb/client/internal/Crypt.java [hunk 1]
```diff
@@ -348,7 +350,9 @@ private void decryptKeys(final MongoCryptContext cryptContext) {
             }
             cryptContext.completeKeyDecryptors();
         } catch (Throwable t) {
-            throw wrapInClientException(t);
+            throw translateInterruptedException(t, "Interrupted while reading")
```
Comment (jyemin) [inline driver-sync/src/main/com/mongodb/client/internal/Crypt.java:353]: Is it necessary to translate interrupted exceptions both here and in the KeyManagementService (which this is calling)?

File: driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java [hunk 2]
```diff
@@ -98,8 +101,21 @@ private void enableHostNameVerification(final SSLSocket socket) {
...
         }
     }
+
+    /**
+     * @return Never.
+     */
+    private static RuntimeException translateInterruptedExceptionAndThrow(final IOException e, final String message) throws IOException,
+            MongoInterruptedException {
+        Optional<MongoInterruptedException> interruptedException = translateInterruptedException(e, message);
+        if (interruptedException.isPresent()) {
```
Comment (jyemin) [inline driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java:115]: Why can't we use `orElseGet` pattern here as in Crypt?

File: driver-sync/src/main/com/mongodb/client/internal/Crypt.java (hunk 1 above)
Comment (stIncMale) [reply driver-sync/src/main/com/mongodb/client/internal/Crypt.java:353]: Only if we want to add messages like "interrupted while connecting/writing/reading" which is what we do for streams, and why I did it this way here. But if it's fine to just have a single message "interrupted while doing IO", or no message at all, I'll be happy to remove the translating code from `Crypt`.

File: driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java (hunk 2 above)
Comment (stIncMale) [reply driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java:115]: We can if we change `KeyManagementService.stream` from `throws IOException` to `throws Exception`. I don't want to do that. Fortunately, if we do https://github.com/mongodb/mongo-java-driver/pull/1204/files#r1337520600, this change in `KeyManagementService` will be gone.

File: driver-sync/src/main/com/mongodb/client/internal/Crypt.java (hunk 1 above)
Comment (stIncMale) [reply driver-sync/src/main/com/mongodb/client/internal/Crypt.java]: Done.

**Review by jyemin — APPROVED** [review]

**Review by vbabanin — APPROVED** [review]
LGTM

**Pull request merged**